centreon/www/include/configuration/configObject/hostgroup/DB-Func.php

```diff
@@ -265,180 +265,631 @@ function multipleHostGroupInDB($hostGroups = array(), $nbrDup = array())
+            :rrdRetention,
+            :comment,
+            :geoCoords,
+            :isActivated
+        )
+    SQL;
+
+    $statement = $pearDB->prepare($request);
+
+    $statement->bindValue(
+        ':name',
+        ! empty($submittedValues['hg_name']) ? $pearDB->escape($submittedValues['hg_name']) : null,
+        \PDO::PARAM_STR
+    );
+
+    $statement->bindValue(
+        ':alias',
+        ! empty($submittedValues['hg_alias']) ? $pearDB->escape($submittedValues['hg_alias']) : null,
+        \PDO::PARAM_STR
+    );
+
+    $statement->bindValue(
+        ':notes',
+        ! empty($submittedValues['hg_notes']) ? $pearDB->escape($submittedValues['hg_notes']) : null,
+        \PDO::PARAM_STR
+    );
+
+    $statement->bindValue(
+        ':notesUrl',
+        ! empty($submittedValues['hg_notes_url']) ? $pearDB->escape($submittedValues['hg_notes_url']) : null,
+        \PDO::PARAM_STR
+    );
+
+    $statement->bindValue(
+        ':actionUrl',
+        ! empty($submittedValues['hg_action_url']) ? $pearDB->escape($submittedValues['hg_action_url']) : null,
+        \PDO::PARAM_STR
+    );
+
+    $statement->bindValue(
+        ':iconImage',
+        ! empty($submittedValues['hg_icon_image']) ? $pearDB->escape($submittedValues['hg_icon_image']) : null,
+        \PDO::PARAM_STR
+    );
+
+    $statement->bindValue(
+        ':mapIconImage',
+        ! empty($submittedValues['hg_map_icon_image']) ? $pearDB->escape($submittedValues['hg_map_icon_image']) : null,
+        \PDO::PARAM_STR
+    );
+
+    $statement->bindValue(
+        ':rrdRetention',
+        ! empty($submittedValues['hg_rrd_retention']) ? $pearDB->escape($submittedValues['hg_rrd_retention']) : null,
+        \PDO::PARAM_STR
+    );
+
+    $statement->bindValue(
+        ':comment',
+        ! empty($submittedValues['hg_comment']) ? $pearDB->escape($submittedValues['hg_comment']) : null,
+        \PDO::PARAM_STR
+    );
+
+    $statement->bindValue(
+        ':geoCoords',
+        ! empty($submittedValues['geo_coords']) ? $pearDB->escape($submittedValues['geo_coords']) : null,
+        \PDO::PARAM_STR
+    );
+
+    $statement->bindValue(
+        ':isActivated',
+        isset($submittedValues['hg_activate']['hg_activate']) && $submittedValues['hg_activate']['hg_activate']
+            ? $submittedValues['hg_activate']['hg_activate']
+            : '0',
+        \PDO::PARAM_STR
+    );
+
+    $statement->execute();
+
+    $statement = $pearDB->query('SELECT MAX(hg_id) FROM hostgroup');
+    $record = $statement->fetch(\PDO::FETCH_ASSOC);
+
+    $centreon->CentreonLogAction->insertLog(
+        object_type: 'hostgroup',
+        object_id: $record['MAX(hg_id)'],
+        object_name: CentreonDB::escape($submittedValues['hg_name']),
+        action_type: 'a',
+        fields: CentreonLogAction::prepareChanges($submittedValues)
+    );
+
+    return ($record["MAX(hg_id)"]);
+}
+
+function insertHostGroup(array $submittedValues): int
+{
+    global $isCloudPlatform;
```

Comment:
Minimize the use of global parameters.
```suggestion
function insertHostGroup(array $submittedValues, bool $isCloudPlatform): int
{
```